modules/Microsoft.ServiceFabric/clusters/.test/common/deploy.test.bicep

```
@@ -0,0 +1,203 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'ms.servicefabric.clusters-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'sfccom'
+
+// =========== //
+// Deployments //
+// =========== //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: location
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+module testDeployment '../../deploy.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name)}-test-${serviceShort}'
+  params: {
+    name: '<<namePrefix>>${serviceShort}001'
+    lock: 'CanNotDelete'
+    tags: {
+      resourceType: 'Service Fabric'
+      clusterName: '<<namePrefix>>${serviceShort}001'
+
+    }
+    addOnFeatures: [
+      'RepairManager'
+      'DnsService'
+      'BackupRestoreService'
+      'ResourceMonitorService'
+    ]
+    maxUnusedVersionsToKeep: 2
+    azureActiveDirectory: {
+      clientApplication: '<<deploymentSpId>>'
+      clusterApplication: 'cf33fea8-b30f-424f-ab73-c48d99e0b222'
+      tenantId: '<<tenantId>>'
+    }
+    certificateCommonNames: {
+      commonNames: [
+        {
+          certificateCommonName: 'certcommon'
+          certificateIssuerThumbprint: '0AC113D5E1D94C401DDEB0EE2B1B96CC130'
+        }
+      ]
+      x509StoreName: ''
+    }
+    clientCertificateCommonNames: [
+      {
+        certificateCommonName: 'clientcommoncert1'
+        certificateIssuerThumbprint: '0AC113D5E1D94C401DDEB0EE2B1B96CC130'
+        isAdmin: false
+      }
+      {
+        certificateCommonName: 'clientcommoncert2'
+        certificateIssuerThumbprint: '0AC113D5E1D94C401DDEB0EE2B1B96CC131'
+        isAdmin: false
+      }
+    ]
+    clientCertificateThumbprints: [
+      {
+        certificateThumbprint: '0AC113D5E1D94C401DDEB0EE2B1B96CC130'
+        isAdmin: false
+      }
+      {
+        certificateThumbprint: '0AC113D5E1D94C401DDEB0EE2B1B96CC131'
+        'isAdmin': false
+      }
+    ]
+    diagnosticsStorageAccountConfig: {
+      blobEndpoint: 'https://adp<<namePrefix>>azsaweux001.blob.core.windows.net/'
```

I think they need to be updated to dependencies (i.e., a storage account)

---

Not sure how to do this as the dependencies have the storage account id

---

No problem. What you need to do is to add a storage account deployment to the dependencies, add an output for its resource id and then update the above line to something like
```suggestion
      blobEndpoint: 'https://${last(split(resourceGroupResources.storageAccountResourceId,' /') )} .blob.core.windows.net/'
```

---

Ready